app.py

```diff
@@ -123,7 +123,17 @@ def skill():
     Handles Skill requests
     '''
     if request.method == 'GET':
-        return jsonify({})
+        try:
+            skill_id = request.args.get('id')
+            if skill_id is not None:
+                skill_id = int(skill_id)
+                if 0 <= skill_id < len(data['skill']):
+                    return jsonify(data['skill'][skill_id]), 200
+                raise ValueError
+            else:
+                return jsonify({})
+        except:
+            return jsonify({'error': 'Invalid request'}), 404
```

Comment:
Maybe 400 here?

Reply:
I have added a commit that implements these proposed changes.